reframe/frontend/cli.py

```diff
@@ -63,6 +63,13 @@ def fmt_deps():
     else:
         node_alloc_scheme = f'flexible (minimum {-check.num_tasks} task(s))'
 
+    hooks = {}
+    for c in inspect.getmro(type(check)):
+        if hasattr(c, '_rfm_pipeline_hooks'):
+            for k, v in c._rfm_pipeline_hooks.items():
+                hooks.setdefault(k, [])
+                hooks[k].extend(v)
```

Comment:
This is too esoteric to be exposed here. I suggest having a function at the class level the returns all the hooks.